experimental/PropertyDDS/packages/property-shared-tree-interop/src/schemaConverter.ts

```diff
@@ -124,17 +113,21 @@ export function convertPSetSchemaToSharedTreeLls(
 
 		const schemaTemplate = PropertyFactory.getTemplate(unprocessedTypeID);
 		if (schemaTemplate === undefined) {
-			throw new Error(`Unknown typeid: ${unprocessedTypeID}`);
+			fail(`Unknown typeid: ${unprocessedTypeID}`);
 		}
 		const dependencies = PropertyTemplate.extractDependencies(
 			schemaTemplate,
 		) as TreeSchemaIdentifier[];
 		for (const dependencyTypeId of dependencies) {
-			const idsSet = getInheritedChildrenForType(inheritedChildrenByType, dependencyTypeId);
-			idsSet.forEach((id) => {
-				if (!referencedTypeIDs.has(dependencyTypeId) && !unprocessedTypeIdsSet.has(id)) {
+			[
+				dependencyTypeId,
+				...getChildrenForType(inheritingChildrenByType, dependencyTypeId),
+			].forEach((id) => {
+				if (
+					!referencedTypeIDs.has(id) &&
+					!unprocessedTypeIds.find((unprocessedTypeId) => unprocessedTypeId === id)
```

Comment:
Is this sufficiently effective?

Reply:
Do you mean performance-wise? The logic should be the same: typeid can be either in `unprocessedTypeIds` or in `referencedTypeIDs` as we unconditionally move it from one list to another. Do you see any difference here?

Reply:
Yes, I meant performance-wise. It is probably ok for medium sized schema. And will consume less memory, I just wanted an opinion.